src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -4696,7 +4762,61 @@ private void compileEntireProject(File sourceDir, Collection<File> recompileJava
                     new String[] { "java" }, true);
             recompileJavaTestSet.addAll(allJavaTestSources);
         }
+    }
 
+    private boolean compileAllClasses(boolean tests, ThreadPoolExecutor executor) throws PluginExecutionException, IOException {
+        return compileAllClasses(this.sourceDirectory, this.testSourceDirectory, packagingType, compileArtifactPaths,
+                testArtifactPaths, outputDirectory, testOutputDirectory, getProjectName(), buildFile, compilerOptions,
+                skipUTs, failedCompilationJavaSources, failedCompilationJavaTests, tests, executor);
+    }
+
+    private boolean compileAllClasses(ProjectModule project, boolean tests, ThreadPoolExecutor executor)
+            throws IOException, PluginExecutionException {
+        return compileAllClasses(project.getSourceDirectory(), project.getTestSourceDirectory(),
+                project.getPackagingType(), project.getCompileArtifacts(), project.getTestArtifacts(),
+                project.getOutputDirectory(), project.getTestOutputDirectory(), project.getProjectName(),
+                project.getBuildFile(), project.getCompilerOptions(), project.skipUTs(),
+                project.failedCompilationJavaSources, project.failedCompilationJavaTests, tests, executor);
+    }
+
+    private boolean compileAllClasses(File sourceDir, File testSourceDir, String packagingType,
```

Comment:
Can you add some javadoc comments to clarify the differences between these methods and `triggerMainModuleCompile`/`triggerUpstreamModuleCompile`?  So that it the distinction is clearer in case future maintenance is needed.